src/Database/Sql/Util/Tables.hs

```diff
@@ -39,18 +39,33 @@ import qualified Data.Text.Lazy as TL
 
 import           Control.Monad.Identity
 import           Control.Monad.Writer
+import           Control.Monad.Reader
 import           Control.Monad.State
 
 import           Database.Sql.Type
 import           Database.Sql.Position
 
 
+-- This describes the usage of a particular table, whether it was read from or
+-- written to. Note that if a table is used in both modes, it will show up
+-- twice as each mode.
+data UsageMode = ReadData | ReadMeta | WriteData | WriteMeta | Unknown
```

Comment:
There's definitely many different ways to represent the idea of table usage with these modes, but for the use cases we were looking at (getting only tables that were read from, or only tables that were writing as part of an ETL), we want the usage mode to act as an index on the usages.